File: app/src/main/java/com/masterjefferson/flockulator/domain/twitter/auth/data/TwitterCredentialStore.java
package com.masterjefferson.flockulator.domain.twitter.auth.data;

import com.masterjefferson.flockulator.domain.twitter.auth.model.TwitterCredentials;

/**
 * ${FILE_NAME}
 * Created by jeff on 9/21/17.
 */

public interface TwitterCredentialStore {
  String getConsumerKey();
  String getConsumerSecret();
  void storeCredentials(TwitterCredentials credentials);
  TwitterCredentials getCredentials();
}

File: app/src/main/java/com/masterjefferson/flockulator/domain/twitter/auth/data/TwitterCredentialStore.kt
package com.masterjefferson.flockulator.domain.twitter.auth.data

import com.masterjefferson.flockulator.domain.twitter.auth.model.TwitterCredentials

/**
 * ${FILE_NAME}
 * Created by jeff on 9/21/17.
 */

interface TwitterCredentialStore {
    val consumerKey: String
    val consumerSecret: String
    val credentials: TwitterCredentials?
    fun storeCredentials(credentials: TwitterCredentials)
}
